vector/src/main/java/im/vector/app/features/voice/AudioWaveformView.kt

```diff
@@ -156,7 +156,9 @@ class AudioWaveformView @JvmOverloads constructor(
             val barHeight = max(fft.value / MAX_FFT * (height - verticalPadding * 2), barMinHeight)
             visibleBarHeights.add(FFT(barHeight, fft.color))
             if (visibleBarHeights.size > maxVisibleBarCount) {
-                visibleBarHeights = visibleBarHeights.subList(visibleBarHeights.size - maxVisibleBarCount, visibleBarHeights.size)
+                visibleBarHeights = mutableListOf<FFT>().apply {
+                    addAll(visibleBarHeights.subList(visibleBarHeights.size - maxVisibleBarCount, visibleBarHeights.size))
+                }
```

Comment:
I understand that we want to keep the last `maxVisibleBarCount` items on the list.

Can it be rewritten like this:

```kotlin
if (visibleBarHeights.size > maxVisibleBarCount) {
        visibleBarHeights = visibleBarHeights.takeLast(maxVisibleBarCount).toMutableList()
        // Or, but maybe less clear
        visibleBarHeights = visibleBarHeights.drop(maxVisibleBarCount - visibleBarHeights.size).toMutableList()
}
```

Alternatively we could drop the first items like this:

```kotlin
            while (visibleBarHeights.size > maxVisibleBarCount) {
                visibleBarHeights.removeAt(0)
            }
```

which has the advantage of not creating a new list. Also it should in fact only drop one item, each time we add an item and make the list too long.

To me those both suggestions are easier to read.

Reply:
Good idea, thanks!